pkgs/development/python-modules/luma-core/default.nix

```
@@ -0,0 +1,43 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  pillow,
+  smbus2,
+  pyftdi,
+  cbor2,
+  rpi-gpio,
+  spidev,
+}:
+buildPythonPackage rec {
+  pname = "luma-core";
+  version = "2.5.1";
+
+  src = fetchPypi {
+    pname = "luma_core";
+    inherit version;
+    sha256 = "sha256-m4j76GiYnsRHfiEP4pk+d0p0Jdfeg3iBQQn5i+MsplM=";
+  };
+
+  build-system = [ setuptools ];
+  pyproject = true;
+
+  propagatedBuildInputs = [
+    setuptools
+    pillow
+    smbus2
+    pyftdi
+    cbor2
+    rpi-gpio
+    spidev
+  ];
+
+  meta = with lib; {
+    description = "A component library providing a Pillow-compatible drawing canvas, and other functionality to support drawing primitives and text-rendering capabilities for small displays on the Raspberry Pi and other single board computers";
+    homepage = "https://luma-core.readthedocs.io/";
+    changelog = "https://github.com/rm-hull/luma.core/blob/${version}/CHANGES.rst";
+    license = licenses.mit;
+    maintainers = with maintainers; [ eymeric ];
+  };
```

```suggestion
  meta = {
    description = "Pillow-compatible drawing canvas library for SBCs";
    longDescription = ''
      A component library providing a Pillow-compatible drawing canvas,
      and other functionality to support drawing primitives and text-rendering
      capabilities for small displays on the Raspberry Pi and other single
      board computers.
    '';
    homepage = "https://luma-core.readthedocs.io/";
    changelog = "https://github.com/rm-hull/luma.core/blob/${version}/CHANGES.rst";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ eymeric ];
  };
```
1. `with lib` is getting removed from everywhere, let's not introduce it back.
2. your `description` is too long and better fits for `longDescription`, I came up with a shorter `description`